File: go/problems/day14_test.go
package problems

import "testing"

var day14 Solvable[int] = Day14{}

func TestDay14_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day14_1")
	result := day14.Solve(input)

	var expected int = 136
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay14_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day14_2")
	result := day14.Solve(input)

	var expected int = 107142
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day14.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day14.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 14", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day14_1");
    const result = solve(input);
    assert.equal(result, 136);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day14_2");
    const result = solve(input);
    assert.equal(result, 107142);
  });
});
